### PR TITLE
fix(pty): render the right margin and combining marks the way a terminal does

### LINKED ISSUE #503: fix(pty): emulate autowrap after a wide grapheme fills the right margin
## Problem

`screen:` is rendered through `github.com/charmbracelet/x/vt`. When a double-width grapheme fills the final two columns, the emulator does not arm pending autowrap. The following printable character is written into the same row and corrupts the wide glyph instead of starting the next row.

## Reproduction

```yaml
- pty:
    shell: true
    command: "printf 日本X; sleep 0.2"
    rows: 5
    cols: 4
- assert:
    screen:
      line: 1
      equals: "日本"
- assert:
    screen:
      line: 2
      equals: "X"
```

A real terminal renders `日本` on line 1 and `X` on line 2. Atago currently renders `日 X` on line 1 (with no second line).

## Scope

- Resolve this in the upstream terminal emulator or carry a safe local workaround.
- Keep the regression in `test/e2e/atago/pty.atago.yaml`; remove `expect_fail` once it passes.
- Cover both a character immediately after an exact-width wide grapheme and a wide grapheme that cannot fit at the right margin.

The existing XFAIL test was added in #480 and continues to run on non-Windows CI.

### LINKED ISSUE #505: fix(pty): preserve combining marks in rendered screen graphemes
## Problem

`screen:` assertions lose combining marks emitted by a TUI. Atago now carries the full `uv.Cell.Content` out of the emulator (the atago-side truncation in #437 is fixed), but the current `github.com/charmbracelet/x/vt` emulator still renders a decomposed grapheme as its base character only.

## Reproduction

```yaml
version: "1"
suite: {name: combining mark}
scenarios:
  - name: screen preserves a decomposed grapheme
    steps:
      - pty:
          shell: true
          command: "printf 'e\\314\\201'; sleep 0.2"
          rows: 5
          cols: 10
      - assert:
          screen:
            line: 1
            equals: "é"
```

The scenario fails: expected `é`, actual `e`.

## Impact

Terminal applications that emit decomposed Unicode cannot be tested exactly with `screen:` or `expect_screen:`. Text matching and `attrs:` matching both observe a different grapheme from what the program wrote.

## Scope

- Confirm the behavior against the current x/vt version and report/fix it upstream, or use a safe local workaround.
- Add the reproduction to the self-hosted PTY suite as an XFAIL until it passes.
- Cover a decomposed base-plus-combining sequence and, if the upstream fix supports it, a multi-code-point grapheme in text and `attrs:` assertions.

This is distinct from #437: that issue was atago discarding an already-complete emulator cell; this one occurs inside the current upstream emulator before atago reads the cell.

### PR DESCRIPTION
**Pull request by nao1215**
## What

Two flaws in the terminal emulator `screen:` is rendered through (`github.com/charmbracelet/x/vt`) changed what an assertion compares against. Both are still present in its latest revision (`v0.0.0-20260816001655`, checked against the emulator directly), so both are corrected on atago's side of the boundary.

**Right margin after a wide character (#503).** Pending wrap is armed only from the LAST column (`atPhantom = awm && x >= width-1`), and a wide character reaches the right margin from the second-to-last one. So after `日本` filled a four-column row the wrap stayed disarmed: `X` was written *inside* the row, overwriting the second half of `本` and blanking its first half (`日 X`), and a wide character that no longer fit was dropped entirely (`日本語` in five columns rendered `日本`).

**Combining marks (#505).** An ASCII character takes the emulator's single-character fast path and is committed to the screen the moment it arrives, so a combining mark that follows can never join it and is dropped — a program writing a decomposed `é` rendered `e`. A non-ASCII base (`か` + dakuten) and a ZWJ emoji sequence go through the emulator's grapheme buffer and were never affected, which is why #437 looked complete.

Fixes #503. Fixes #505.

## How

`screenWriter` sits between the sanitized transcript and the emulator. It decodes the transcript with **the same library the emulator parses it with** (`x/ansi`), so the units it sees are the units the emulator will see, and it hands everything that needs no correction over in batches — ordinary output reaches the emulator exactly as before.

- A wide character that reaches the margin arms the wrap; one that no longer fits is wrapped rather than dropped. The wrap is injected as CR+LF, so scroll regions, origin mode, and scrolling at the bottom margin all behave as the program set them, and both codes clear the emulator's own pending-wrap state so a wrap can never fire twice. A cursor move between the armed wrap and the next character cancels it, the way a terminal cancels its own.
- DECAWM is tracked (`CSI ? 7 h/l`, and the RIS that restores it): with autowrap off a terminal keeps overwriting the last column, so every correction switches off with it.
- A cluster whose base is ASCII is re-read as a whole grapheme cluster, and after the base is written its cell content is replaced with the full cluster. Only the base is handed to the emulator — a mark arriving while pending wrap is armed makes it index a row (and scroll the screen when the base sat on the last one) for a grapheme that moves the cursor nowhere.

A cluster that is not valid UTF-8 is left alone: the segmenter will carry a stray continuation byte along with the character before it, and the emulator drops invalid UTF-8 rather than printing it. `FuzzRenderScreen` found that within a second of the first version; its input is committed as a seed.

## Tests

- `TestRenderScreen_WideCharacterAutowrap`: the two broken cases plus what must NOT change — an explicit newline after a filled row does not skip one, cursor addressing cancels the pending wrap, `DECAWM` off keeps the overwrite behavior, narrow autowrap still wraps.
- `TestRenderScreen_PreservesCombiningMarks`: one mark, two marks, a mark at the last column, a mark followed by another character, and the precomposed / wide-base spellings that were already right.
- `test/e2e/atago/pty.atago.yaml`: the wide-character scenario was an `expect_fail` carried since #480 — it is now an ordinary passing scenario, joined by the "no longer fits" case and a decomposed-grapheme one. The expected text uses explicit `\u` escapes on both sides, because a screen assertion compares the bytes the program wrote and a decomposed grapheme is not the same text as its precomposed spelling.
- `FuzzRenderScreen`: 5 minutes, ~3.1M execs, no failures.
- Rendering a 1.4 MB ASCII transcript costs ~10% more, a 2 MB CJK one ~3%; the emulator itself dominates both.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved terminal rendering for wide characters at line boundaries and during overflow wrapping.
  - Preserved combining marks and decomposed characters correctly in displayed text.
  - Fixed cursor, newline, and autowrap behavior for more accurate screen output.
  - Improved rendering consistency across resized terminal transcript segments.

- **Tests**
  - Added coverage for wide-character wrapping, combining marks, and related terminal edge cases.

- **Documentation**
  - Updated behavior-spec documentation to reflect the expanded terminal rendering coverage.

### CODE REVIEW

File: internal/runner/ptyrun/screen.go [hunk 1]
```diff
@@ -115,16 +108,19 @@ func renderScreenCells(transcript []byte, p *spec.PTY, resizes []screenResize) (
...
 		cut := min(max(cuts[i], at), len(sanitized))
-		writeTranscript(term, sanitized[at:cut])
+		w.write(sanitized[at:cut])
 		// Resize takes width (cols) first; getting that backwards silently
 		// transposes every frame after a resize.
 		term.Resize(r.cols, r.rows)
 		at = cut
 	}
-	writeTranscript(term, sanitized[at:])
+	w.write(sanitized[at:])
```
[the remark this answers is not in the document]
Comment (nao1215) [reply internal/runner/ptyrun/screen.go]: Fixed in 7dbe650. The transcript is now written in one pass with the resizes handed to the writer as offsets, and each one is applied at a boundary between whole units — so an offset that lands inside a grapheme cluster can no longer separate a base character from its mark (nor land inside an escape sequence). `TestRenderScreenResized_SplitGraphemeCluster` puts a resize at offset 1 of `éX` and asserts the mark survives; it fails on the previous commit with `eX`.

File: internal/runner/ptyrun/screenwrite.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	// it first.
+	w.flush()
+	if w.pending {
+		// A sequence that moved the cursor since the wrap was armed cancels it,
+		// the way a terminal cancels its own pending wrap.
+		if w.term.CursorPosition() == w.pendingAt {
+			w.writeNow(crlf)
+		}
+		w.pending = false
+	}
```
[the remark this answers is not in the document]
Comment (nao1215) [reply internal/runner/ptyrun/screenwrite.go:234]: Fixed in 7dbe650. `pending` is now cleared when a cursor-moving unit is decoded — BS/HT/LF/VT/FF/CR, IND/NEL/RI/DECSC/DECRC/RIS, and the CSI finals that reposition the cursor — with the position comparison kept as a backstop for a move that list does not name. SGR, mode sets, and status reports are deliberately not in it, so a TUI that colors its output between two characters keeps the wrap that was armed before it. `TestRenderScreen_PendingWrapCancelledByCursorMove` covers your exact sequence and fails on the previous commit with `日本\nX`.
